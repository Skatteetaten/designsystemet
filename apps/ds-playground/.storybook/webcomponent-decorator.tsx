import { Decorator } from '@storybook/react';
import { createRoot, Root } from 'react-dom/client';

export interface WebcompAttributes {
  [key: string]: string | undefined;
}
export const webComponent: Decorator = function (Story, context) {
  const Name = context.parameters['customElementName'] ?? 'ds-customelement';
  class WebComponent extends HTMLElement {
    mountPoint!: HTMLElement;
    props!: WebcompAttributes; // Frivillige attributes som custom-elementet kan ta imot - de oversettes til props
    root!: Root;

    connectedCallback(): void {
      this.mountReactComponent();
    }

    disconnectedCallback(): void {
      // setTimeout for å gjøre unmount async. Ellers får vi en feilmelding
      // Warning: Attempted to synchronously unmount a root while React was already rendering.
      // React cannot finish unmounting the root until the current render has completed,
      // which may lead to a race condition.
      setTimeout(() => {
        this.root.unmount();
      });
    }

    // henter custom-elementets attributer - de assigneres til props lengre ned
    get value(): WebcompAttributes {
      return Array.from(this.attributes).reduce(
        (previousValue: { [key: string]: string }, currentValue) => {
          previousValue[currentValue.name] = currentValue.value;
          return previousValue;
        },
        {}
      ) as WebcompAttributes;
    }

    // håndtering av events - disse skal sendes videre til komponentene gjennom bruk av EventForwarder wrappper
    eventDispatcher = (event: Event): void => {
      this.dispatchEvent(event);
    };

    subscribeDispatcher = (
      eventname: string,
      callback: (event: Event) => void
    ): void => {
      this.addEventListener(eventname, (e) => {
        return callback(e);
      });
    };

    mountReactComponent(): void | null {
      this.props = this.value;
      this.mountPoint = document.createElement('div');

      const shadowRoot = this.attachShadow({ mode: 'open' });

      shadowRoot.appendChild(this.mountPoint);

      if (import.meta.env.MODE === 'development') {
        for (const tag of document.head.querySelectorAll('style')) {
          const clone = tag.cloneNode(true);
          shadowRoot.appendChild(clone);
        }
      } else {
        const cssLink = document.createElement('link');
        cssLink.href = 'main.css';
        cssLink.rel = 'stylesheet';
        shadowRoot.appendChild(cssLink);
      }

      this.root = createRoot(this.mountPoint);
      this.root.render(<Story {...this.props} />);
    }
  }

  if (!window.customElements.get(Name)) {
    window.customElements.define(Name, WebComponent);
  }

  return <Name />;
};
