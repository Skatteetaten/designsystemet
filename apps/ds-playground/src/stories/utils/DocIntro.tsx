import { ReactNode } from 'react';

import { Link } from '@skatteetaten/ds-buttons';
import { Paragraph } from '@skatteetaten/ds-typography';

interface DocIntroProps {
  name: string;
  url: string;
}

export const DocIntro = ({ name, url }: DocIntroProps): ReactNode => {
  return (
    <Paragraph variant={'ingress'}>
      {`Se `}
      <Link href={url}>{name}</Link>
      {` p√• stil og tone for eksempler og
        retningslinjer for denne komponenten.`}
    </Paragraph>
  );
};
