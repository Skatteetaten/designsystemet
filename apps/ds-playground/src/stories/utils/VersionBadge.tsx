import { ReactNode } from 'react';

interface VersionBadgeProps {
  packageName: string;
}

export const VersionBadge = ({ packageName }: VersionBadgeProps): ReactNode => {
  let color = 'green';

  let version =
    import.meta.env.DS_VERSIONS[`${packageName.replaceAll('-', '_')}`] ?? '..';
  if (import.meta.env.MODE === 'development') {
    color = 'red';
    version = 'developing';
  }
  if (import.meta.env['STORYBOOK_BUILD_VERSION']?.includes('SNAPSHOT')) {
    color = 'blue';
  }

  const normalizedPackageName = packageName.replaceAll('-', '--');
  const normalizedVersion = version.replaceAll('-', '--').replaceAll('_', '__');

  return (
    <img
      src={`https://img.shields.io/badge/%40skatteetaten%2Fds--${normalizedPackageName}-${normalizedVersion}-${color}?logo=npm`}
      alt={`@skatteetaten/ds-${packageName}@${version}`}
    />
  );
};
