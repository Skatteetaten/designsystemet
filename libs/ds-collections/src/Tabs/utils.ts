export const valueRegex = /^[A-Za-z][\w\-.:]*$/;
