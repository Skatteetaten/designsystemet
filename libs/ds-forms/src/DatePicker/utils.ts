import { format, isValid, parse } from 'date-fns';

export const formatDateForInput = (
  dateFormat: string,
  date?: Date | null
): string => {
  return !isValid(date) ? '' : format(date as Date, dateFormat);
};

const COMMON_INPUT_FORMATS = [
  'dd.MM.yy',
  'dd/MM/yy',
  'dd-MM-yy',
  'dd.MM.yyyy',
  'dd/MM/yyyy',
  'dd-MM-yyyy',
  'ddMM',
  'ddMMyy',
  'ddMMyyyy',
];

export const parseDateFromInput = (
  value: string,
  dateFormat: string
): Date | null => {
  const dateFormats = Array.from(
    new Set([...COMMON_INPUT_FORMATS, dateFormat])
  );

  for (const format of dateFormats) {
    const date = parse(value, format, new Date());
    if (isValid(date)) {
      return date;
    }
  }

  return null;
};
