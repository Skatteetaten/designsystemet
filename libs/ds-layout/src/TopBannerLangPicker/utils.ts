import { Languages } from '@skatteetaten/ds-core-utils';

import { TopBannerLangPickerProps } from './TopBannerLangPicker.types';

export const convertLocaleToLang = (locale: Languages): string => {
  return locale.substring(0, locale.indexOf('_'));
};

export type LanguageItem = {
  lang: string;
  displayName: string;
};

export type LanguageItems = {
  [key: string]: LanguageItem;
};

export const isLanguages = (value: unknown): value is Languages => {
  return Object.values(Languages).includes(value as Languages);
};

export const getCurrentLanguages = (
  includeSami: boolean,
  additionalLanguages: TopBannerLangPickerProps['additionalLanguages'] = []
): LanguageItems => {
  const defaultLanguages: LanguageItems = {
    nb: {
      lang: 'nb',
      displayName: 'Bokmål',
    },
    nn: {
      lang: 'nn',
      displayName: 'Nynorsk',
    },
    en: {
      lang: 'en',
      displayName: 'English',
    },
  };
  const se = {
    se: {
      lang: 'se',
      displayName: 'Sámegiella',
    },
  };

  const additionalLanguagesAsObject =
    additionalLanguages?.reduce<LanguageItems>((acc, curr) => {
      acc[curr.lang] = curr;
      return acc;
    }, {});

  return includeSami
    ? { ...defaultLanguages, ...se, ...additionalLanguagesAsObject }
    : { ...defaultLanguages, ...additionalLanguagesAsObject };
};
