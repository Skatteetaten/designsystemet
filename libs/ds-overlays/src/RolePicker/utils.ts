import { formatter } from '@skatteetaten/ds-core-utils';

export const formatOrganisationNumber = (
  organisationNumber: string
): string => {
  const result = formatter({
    type: 'organisationNumber',
    value: organisationNumber,
  });

  return result.value;
};

export const formatNationalIdentityNumber = (
  nationalIdentityNumber: string
): string => {
  const result = formatter({
    type: 'nationalIdentityNumber',
    value: nationalIdentityNumber,
  });

  return result.value;
};
