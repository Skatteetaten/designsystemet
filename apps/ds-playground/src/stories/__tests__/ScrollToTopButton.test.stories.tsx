import { ScrollToTopButton } from '@skatteetaten/ds-buttons';
import { ExternalLayout, dsI18n } from '@skatteetaten/ds-core-utils';
import { StoryFn, Meta, StoryObj } from '@storybook/react';
import { expect, userEvent, within } from '@storybook/test';

import { wrapper } from './testUtils/storybook.testing.utils';
import { webComponent } from '../../../.storybook/webcomponent-decorator';

const defaultButtonText = 'Til toppen';

const meta = {
  component: ScrollToTopButton,
  title: 'Tester/ScrollToTopButton',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    classNames: {
      table: { disable: true },
    },
    visibilityThreshold: {
      table: { disable: true },
    },
    shadowRootNode: { table: { disable: true } },
    children: { table: { disable: true } },
  },
  parameters: {
    backgrounds: {
      default: 'grey',
    },
  },
} satisfies Meta<typeof ScrollToTopButton>;
export default meta;
type Story = StoryObj<typeof meta>;

const Template: StoryFn<typeof ScrollToTopButton> = (args) => (
  <div className={'height100vh'}>
    <main className={'scrollToTopContainer'} tabIndex={-1}>
      <ExternalLayout />
      <ScrollToTopButton {...args} />
    </main>
  </div>
);

const defaultArgs = {
  visibilityThreshold: 0,
};

export const WithRef = {
  render: Template,
  name: 'With Ref (FA1)',
  args: {
    ...defaultArgs,
    ref: (instance: HTMLButtonElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },
  argTypes: {
    ref: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: {
      disable: true,
    },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    expect(canvas.getByRole('button')).toHaveAttribute(
      'id',
      'dummyIdForwardedFromRef'
    );
  },
} satisfies Story;

export const WithAttributes = {
  render: Template,
  name: 'With Attributes(FA2-5)',
  args: {
    ...defaultArgs,
    id: 'htmlId',
    className: 'dummyClassname',
    lang: 'nb',
    'data-testid': '123ID',
  },
  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const scrollToTopButton = canvas.getByRole('button');
    await expect(scrollToTopButton).toHaveAttribute('id', 'htmlId');
    await expect(scrollToTopButton).toHaveAttribute('lang', 'nb');
    await expect(scrollToTopButton).toHaveAttribute('data-testid', '123ID');
    // eslint-disable-next-line testing-library/no-node-access
    const container = canvasElement.querySelector(
      `${wrapper} > div > main > div:nth-child(2)`
    );
    await expect(container).toHaveClass('dummyClassname');
  },
} satisfies Story;

export const WithCustomClassNames = {
  render: Template,
  name: 'With Custom ClassNames (FA3)',
  args: {
    ...defaultArgs,
    classNames: {
      container: 'dummyClassname',
      button: 'dummyClassname',
      iconContainer: 'dummyClassname',
      icon: 'dummyClassname',
      label: 'dummyClassname',
    },
  },
  argTypes: {
    classNames: {
      table: { disable: false },
    },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    /* eslint-disable testing-library/no-node-access */
    const container = canvasElement.querySelector(
      `${wrapper} > div > main > div:nth-child(2)`
    );
    const button = canvas.getByRole('button');
    const iconContainer = button.querySelector('div');
    /* eslint-disable testing-library/no-node-access */
    const icon = canvas.getByRole('img', { hidden: true });
    const label = canvas.getByText(
      dsI18n.t('ds_buttons:scrolltotopbutton.Title')
    );
    await expect(container).toHaveClass('dummyClassname');
    await expect(button).toHaveClass('dummyClassname');
    await expect(iconContainer).toHaveClass('dummyClassname');
    await expect(icon).toHaveClass('dummyClassname');
    await expect(label).toHaveClass('dummyClassname');
  },
} satisfies Story;

export const Defaults = {
  render: Template,
  name: 'Defaults (A1, A3, B4)',
  args: {
    ...defaultArgs,
  },
  argTypes: {
    visibilityThreshold: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: {
      hover: `${wrapper} > div > main > div:nth-child(2) > button`,
      focus: `${wrapper} > div > main > div:nth-child(2) > button`,
      click: `${wrapper} > div > main > div:nth-child(2) > button`,
    },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const scrollToTopButton = canvas.getByText(defaultButtonText);
    await expect(scrollToTopButton).toBeInTheDocument();
    await expect(canvas.getByRole('button')).toHaveAttribute('type', 'button');
  },
} satisfies Story;

export const WithChildren = {
  render: Template,
  name: 'With Children (A2)',
  args: {
    ...defaultArgs,
    children: 'dummy string',
  },
  argTypes: {
    children: { table: { disable: false } },
  },
  play: async ({ args, canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    await expect(canvas.getByText(args.children ?? '')).toBeInTheDocument();
  },
} satisfies Story;

export const WithMobileScreen = {
  render: Template,
  name: 'With Small Screen (A5)',
  args: {
    ...defaultArgs,
  },
  parameters: {
    viewport: {
      defaultViewport: '--breakpoint-xs',
    },
  },
} satisfies Story;

export const WithWideScreen = {
  render: Template,
  name: 'With Wide Screen (A6)',
  args: {
    ...defaultArgs,
  },
  parameters: {
    viewport: {
      defaultViewport: '--breakpoint-xl',
    },
  },
} satisfies Story;

export const WithVisibilityThreshold = {
  render: Template,
  name: 'With VisibilityThreshold and Scrolling (A7, A8)',
  args: {
    ...defaultArgs,
    visibilityThreshold: 3,
  },
  argTypes: {
    visibilityThreshold: { table: { disable: false } },
  },
  parameters: {
    viewport: {
      defaultViewport: '--breakpoint-xl',
    },
    imageSnapshot: {
      scroll: {
        yPixels: 10,
      },
    },
  },
} satisfies Story;

const TemplateWithShadowDom: StoryFn<typeof ScrollToTopButton> = (args) => {
  // eslint-disable-next-line testing-library/no-node-access
  const element = document.querySelector('scrolltotop-customelement');
  const shadowRoot = element?.shadowRoot;
  return (
    <div className={'height100vh'}>
      <main className={'scrollToTopContainer'} tabIndex={-1}>
        <ExternalLayout />
        <ScrollToTopButton {...args} shadowRootNode={shadowRoot ?? undefined} />
      </main>
    </div>
  );
};

export const WithShadowDom = {
  render: TemplateWithShadowDom,
  name: 'With ShadowDom (B4)',
  args: {
    ...defaultArgs,
  },
  decorators: [webComponent],
  parameters: {
    imageSnapshot: {
      disable: true,
    },
    customElementName: 'scrolltotop-customelement',
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    //button finnes ikke utenfor shadowDom
    await expect(canvas.queryByRole('button')).not.toBeInTheDocument();
    const customElement = canvasElement.querySelector(
      `scrolltotop-customelement`
    );
    await expect(customElement).toBeInTheDocument();
    const button =
      customElement?.shadowRoot &&
      customElement.shadowRoot.querySelector('button');

    await expect(button).toBeInTheDocument();
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    await userEvent.click(button!);

    const main =
      customElement?.shadowRoot &&
      customElement.shadowRoot.querySelector('main:focus');
    await expect(main).toBeInTheDocument();
  },
} satisfies Story;
