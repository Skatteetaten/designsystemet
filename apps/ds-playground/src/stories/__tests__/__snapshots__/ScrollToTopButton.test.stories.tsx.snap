// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/ScrollToTopButton Defaults play-test 1`] = `
<div data-test-block="true">
  <div class="height100vh">
    <main class="scrollToTopContainer"
          tabindex="-1"
    >
      <div class="_layout_179rg_1 ">
        <style>
          div.css-1gh4yjl { max-width: none}
        </style>
        <p>
          <strong>
            Responsive container
          </strong>
        </p>
        <article class="_article_179rg_12">
          <p>
            <strong>
              Responsive article
            </strong>
          </p>
          <p>
            (uses --semantic-responsive-article)
          </p>
        </article>
        <p>
          (uses --semantic-responsive-container)
        </p>
      </div>
      <div class="_container_gchs5_1  ">
        <button class="_scrollToTopButton_gchs5_6 _scrollToTopButton_visible_gchs5_27 "
                type="button"
        >
          <div class="_scrollToTopButtonIconContainer_gchs5_48 ">
            <svg class="_systemIcon_6zbcy_2 _systemIcon_large_6zbcy_13 _scrollToTopButtonIconContainerIcon_gchs5_56 "
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M11.1 17.6h1.8v1.1h-1.8v-1.1zm0-3.3h1.8v2.2h-1.8v-2.2zm0 6.2h1.8v.5h-1.8v-.5zm-4.9-7.3 5-5v5.1H13V8.2l5 4.9 1.3-1.3-7.3-7-7.1 7.1 1.3 1.3zM21 3H3v1.8h18V3z">
              </path>
            </svg>
          </div>
          <div class="_scrollToTopButtonLabel_gchs5_70 ">
            Til toppen
          </div>
        </button>
      </div>
    </main>
  </div>
</div>
`;

exports[`Tester/ScrollToTopButton WithAttributes play-test 1`] = `
<div data-test-block="true">
  <div class="height100vh">
    <main class="scrollToTopContainer"
          tabindex="-1"
    >
      <div class="_layout_179rg_1 ">
        <style>
          div.css-1gh4yjl { max-width: none}
        </style>
        <p>
          <strong>
            Responsive container
          </strong>
        </p>
        <article class="_article_179rg_12">
          <p>
            <strong>
              Responsive article
            </strong>
          </p>
          <p>
            (uses --semantic-responsive-article)
          </p>
        </article>
        <p>
          (uses --semantic-responsive-container)
        </p>
      </div>
      <div class="_container_gchs5_1 dummyClassname ">
        <button id="htmlId"
                class="_scrollToTopButton_gchs5_6 _scrollToTopButton_visible_gchs5_27 "
                lang="nb"
                type="button"
                data-testid="123ID"
        >
          <div class="_scrollToTopButtonIconContainer_gchs5_48 ">
            <svg class="_systemIcon_6zbcy_2 _systemIcon_large_6zbcy_13 _scrollToTopButtonIconContainerIcon_gchs5_56 "
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M11.1 17.6h1.8v1.1h-1.8v-1.1zm0-3.3h1.8v2.2h-1.8v-2.2zm0 6.2h1.8v.5h-1.8v-.5zm-4.9-7.3 5-5v5.1H13V8.2l5 4.9 1.3-1.3-7.3-7-7.1 7.1 1.3 1.3zM21 3H3v1.8h18V3z">
              </path>
            </svg>
          </div>
          <div class="_scrollToTopButtonLabel_gchs5_70 ">
            Til toppen
          </div>
        </button>
      </div>
    </main>
  </div>
</div>
`;

exports[`Tester/ScrollToTopButton WithChildren play-test 1`] = `
<div data-test-block="true">
  <div class="height100vh">
    <main class="scrollToTopContainer"
          tabindex="-1"
    >
      <div class="_layout_179rg_1 ">
        <style>
          div.css-1gh4yjl { max-width: none}
        </style>
        <p>
          <strong>
            Responsive container
          </strong>
        </p>
        <article class="_article_179rg_12">
          <p>
            <strong>
              Responsive article
            </strong>
          </p>
          <p>
            (uses --semantic-responsive-article)
          </p>
        </article>
        <p>
          (uses --semantic-responsive-container)
        </p>
      </div>
      <div class="_container_gchs5_1  ">
        <button class="_scrollToTopButton_gchs5_6 _scrollToTopButton_visible_gchs5_27 "
                type="button"
        >
          <div class="_scrollToTopButtonIconContainer_gchs5_48 ">
            <svg class="_systemIcon_6zbcy_2 _systemIcon_large_6zbcy_13 _scrollToTopButtonIconContainerIcon_gchs5_56 "
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M11.1 17.6h1.8v1.1h-1.8v-1.1zm0-3.3h1.8v2.2h-1.8v-2.2zm0 6.2h1.8v.5h-1.8v-.5zm-4.9-7.3 5-5v5.1H13V8.2l5 4.9 1.3-1.3-7.3-7-7.1 7.1 1.3 1.3zM21 3H3v1.8h18V3z">
              </path>
            </svg>
          </div>
          <div class="_scrollToTopButtonLabel_gchs5_70 ">
            dummy string
          </div>
        </button>
      </div>
    </main>
  </div>
</div>
`;

exports[`Tester/ScrollToTopButton WithCustomClassNames play-test 1`] = `
<div data-test-block="true">
  <div class="height100vh">
    <main class="scrollToTopContainer"
          tabindex="-1"
    >
      <div class="_layout_179rg_1 ">
        <style>
          div.css-1gh4yjl { max-width: none}
        </style>
        <p>
          <strong>
            Responsive container
          </strong>
        </p>
        <article class="_article_179rg_12">
          <p>
            <strong>
              Responsive article
            </strong>
          </p>
          <p>
            (uses --semantic-responsive-article)
          </p>
        </article>
        <p>
          (uses --semantic-responsive-container)
        </p>
      </div>
      <div class="_container_gchs5_1  dummyClassname">
        <button class="_scrollToTopButton_gchs5_6 _scrollToTopButton_visible_gchs5_27 dummyClassname"
                type="button"
        >
          <div class="_scrollToTopButtonIconContainer_gchs5_48 dummyClassname">
            <svg class="_systemIcon_6zbcy_2 _systemIcon_large_6zbcy_13 _scrollToTopButtonIconContainerIcon_gchs5_56 dummyClassname"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M11.1 17.6h1.8v1.1h-1.8v-1.1zm0-3.3h1.8v2.2h-1.8v-2.2zm0 6.2h1.8v.5h-1.8v-.5zm-4.9-7.3 5-5v5.1H13V8.2l5 4.9 1.3-1.3-7.3-7-7.1 7.1 1.3 1.3zM21 3H3v1.8h18V3z">
              </path>
            </svg>
          </div>
          <div class="_scrollToTopButtonLabel_gchs5_70 dummyClassname">
            Til toppen
          </div>
        </button>
      </div>
    </main>
  </div>
</div>
`;

exports[`Tester/ScrollToTopButton WithMobileScreen smoke-test 1`] = `
<div data-test-block="true">
  <div class="height100vh">
    <main class="scrollToTopContainer"
          tabindex="-1"
    >
      <div class="_layout_179rg_1 ">
        <style>
          div.css-1gh4yjl { max-width: none}
        </style>
        <p>
          <strong>
            Responsive container
          </strong>
        </p>
        <article class="_article_179rg_12">
          <p>
            <strong>
              Responsive article
            </strong>
          </p>
          <p>
            (uses --semantic-responsive-article)
          </p>
        </article>
        <p>
          (uses --semantic-responsive-container)
        </p>
      </div>
      <div class="_container_gchs5_1  ">
        <button class="_scrollToTopButton_gchs5_6 _scrollToTopButton_visible_gchs5_27 "
                type="button"
        >
          <div class="_scrollToTopButtonIconContainer_gchs5_48 ">
            <svg class="_systemIcon_6zbcy_2 _systemIcon_large_6zbcy_13 _scrollToTopButtonIconContainerIcon_gchs5_56 "
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M11.1 17.6h1.8v1.1h-1.8v-1.1zm0-3.3h1.8v2.2h-1.8v-2.2zm0 6.2h1.8v.5h-1.8v-.5zm-4.9-7.3 5-5v5.1H13V8.2l5 4.9 1.3-1.3-7.3-7-7.1 7.1 1.3 1.3zM21 3H3v1.8h18V3z">
              </path>
            </svg>
          </div>
          <div class="_scrollToTopButtonLabel_gchs5_70 ">
            Til toppen
          </div>
        </button>
      </div>
    </main>
  </div>
</div>
`;

exports[`Tester/ScrollToTopButton WithNotScrollToMain play-test 1`] = `
<div data-test-block="true">
  <div class="height200vh">
    <div class="scrollToTopMainPusher">
      Mainpusher
    </div>
    <main class="scrollToTopContainer"
          tabindex="-1"
    >
      <div class="_layout_179rg_1 ">
        <style>
          div.css-1gh4yjl { max-width: none}
        </style>
        <p>
          <strong>
            Responsive container
          </strong>
        </p>
        <article class="_article_179rg_12">
          <p>
            <strong>
              Responsive article
            </strong>
          </p>
          <p>
            (uses --semantic-responsive-article)
          </p>
        </article>
        <p>
          (uses --semantic-responsive-container)
        </p>
      </div>
      <div class="_container_gchs5_1  ">
        <button class="_scrollToTopButton_gchs5_6 _scrollToTopButton_visible_gchs5_27 "
                type="button"
        >
          <div class="_scrollToTopButtonIconContainer_gchs5_48 ">
            <svg class="_systemIcon_6zbcy_2 _systemIcon_large_6zbcy_13 _scrollToTopButtonIconContainerIcon_gchs5_56 "
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M11.1 17.6h1.8v1.1h-1.8v-1.1zm0-3.3h1.8v2.2h-1.8v-2.2zm0 6.2h1.8v.5h-1.8v-.5zm-4.9-7.3 5-5v5.1H13V8.2l5 4.9 1.3-1.3-7.3-7-7.1 7.1 1.3 1.3zM21 3H3v1.8h18V3z">
              </path>
            </svg>
          </div>
          <div class="_scrollToTopButtonLabel_gchs5_70 ">
            Til toppen
          </div>
        </button>
      </div>
    </main>
  </div>
</div>
`;

exports[`Tester/ScrollToTopButton WithRef play-test 1`] = `
<div data-test-block="true">
  <div class="height100vh">
    <main class="scrollToTopContainer"
          tabindex="-1"
    >
      <div class="_layout_179rg_1 ">
        <style>
          div.css-1gh4yjl { max-width: none}
        </style>
        <p>
          <strong>
            Responsive container
          </strong>
        </p>
        <article class="_article_179rg_12">
          <p>
            <strong>
              Responsive article
            </strong>
          </p>
          <p>
            (uses --semantic-responsive-article)
          </p>
        </article>
        <p>
          (uses --semantic-responsive-container)
        </p>
      </div>
      <div class="_container_gchs5_1  ">
        <button class="_scrollToTopButton_gchs5_6 _scrollToTopButton_visible_gchs5_27 "
                type="button"
                id="dummyIdForwardedFromRef"
        >
          <div class="_scrollToTopButtonIconContainer_gchs5_48 ">
            <svg class="_systemIcon_6zbcy_2 _systemIcon_large_6zbcy_13 _scrollToTopButtonIconContainerIcon_gchs5_56 "
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M11.1 17.6h1.8v1.1h-1.8v-1.1zm0-3.3h1.8v2.2h-1.8v-2.2zm0 6.2h1.8v.5h-1.8v-.5zm-4.9-7.3 5-5v5.1H13V8.2l5 4.9 1.3-1.3-7.3-7-7.1 7.1 1.3 1.3zM21 3H3v1.8h18V3z">
              </path>
            </svg>
          </div>
          <div class="_scrollToTopButtonLabel_gchs5_70 ">
            Til toppen
          </div>
        </button>
      </div>
    </main>
  </div>
</div>
`;

exports[`Tester/ScrollToTopButton WithShadowDom play-test 1`] = `
<div data-test-block="true">
  <scrolltotop-customelement>
  </scrolltotop-customelement>
</div>
`;

exports[`Tester/ScrollToTopButton WithVisibilityThreshold smoke-test 1`] = `
<div data-test-block="true">
  <div class="height100vh">
    <main class="scrollToTopContainer"
          tabindex="-1"
    >
      <div class="_layout_179rg_1 ">
        <style>
          div.css-1gh4yjl { max-width: none}
        </style>
        <p>
          <strong>
            Responsive container
          </strong>
        </p>
        <article class="_article_179rg_12">
          <p>
            <strong>
              Responsive article
            </strong>
          </p>
          <p>
            (uses --semantic-responsive-article)
          </p>
        </article>
        <p>
          (uses --semantic-responsive-container)
        </p>
      </div>
      <div class="_container_gchs5_1  ">
        <button class="_scrollToTopButton_gchs5_6  "
                type="button"
        >
          <div class="_scrollToTopButtonIconContainer_gchs5_48 ">
            <svg class="_systemIcon_6zbcy_2 _systemIcon_large_6zbcy_13 _scrollToTopButtonIconContainerIcon_gchs5_56 "
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M11.1 17.6h1.8v1.1h-1.8v-1.1zm0-3.3h1.8v2.2h-1.8v-2.2zm0 6.2h1.8v.5h-1.8v-.5zm-4.9-7.3 5-5v5.1H13V8.2l5 4.9 1.3-1.3-7.3-7-7.1 7.1 1.3 1.3zM21 3H3v1.8h18V3z">
              </path>
            </svg>
          </div>
          <div class="_scrollToTopButtonLabel_gchs5_70 ">
            Til toppen
          </div>
        </button>
      </div>
    </main>
  </div>
</div>
`;

exports[`Tester/ScrollToTopButton WithWideScreen smoke-test 1`] = `
<div data-test-block="true">
  <div class="height100vh">
    <main class="scrollToTopContainer"
          tabindex="-1"
    >
      <div class="_layout_179rg_1 ">
        <style>
          div.css-1gh4yjl { max-width: none}
        </style>
        <p>
          <strong>
            Responsive container
          </strong>
        </p>
        <article class="_article_179rg_12">
          <p>
            <strong>
              Responsive article
            </strong>
          </p>
          <p>
            (uses --semantic-responsive-article)
          </p>
        </article>
        <p>
          (uses --semantic-responsive-container)
        </p>
      </div>
      <div class="_container_gchs5_1  ">
        <button class="_scrollToTopButton_gchs5_6 _scrollToTopButton_visible_gchs5_27 "
                type="button"
        >
          <div class="_scrollToTopButtonIconContainer_gchs5_48 ">
            <svg class="_systemIcon_6zbcy_2 _systemIcon_large_6zbcy_13 _scrollToTopButtonIconContainerIcon_gchs5_56 "
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M11.1 17.6h1.8v1.1h-1.8v-1.1zm0-3.3h1.8v2.2h-1.8v-2.2zm0 6.2h1.8v.5h-1.8v-.5zm-4.9-7.3 5-5v5.1H13V8.2l5 4.9 1.3-1.3-7.3-7-7.1 7.1 1.3 1.3zM21 3H3v1.8h18V3z">
              </path>
            </svg>
          </div>
          <div class="_scrollToTopButtonLabel_gchs5_70 ">
            Til toppen
          </div>
        </button>
      </div>
    </main>
  </div>
</div>
`;
